corehq/form_processor/submission_post.py

```diff
@@ -219,12 +232,25 @@ def run(self):
         report_submission_usage(self.domain)
         failure_response = self._handle_basic_failure_modes()
         if failure_response:
-            return FormProcessingResult(failure_response, None, [], [], 'known_failures')
+            return FormProcessingResponse(failure_response, None, [], [], 'known_failures')
+
+        try:
+            # decode xml to ensure it can be decoded to string when and if needed in pre processing steps
+            instance_xml = self.instance.decode() if isinstance(self.instance, bytes) else self.instance
+        except UnicodeDecodeError:
+            xform_context = SubmissionFormContext(instance_xml=self.instance)
```

Comment:
I don't think you can do this here since this the XML may be encoded with something other than `utf8` but will still be valid XML.